app/models/UserProducts.scala

```diff
@@ -0,0 +1,11 @@
+package models
+
+import play.api.libs.json.{Json, OFormat}
+
+case class UserProducts(name: String,
+                        quantity: Option[Int],
+                        price: Option[Double])
+
+object UserProducts {
+  implicit val format: OFormat[UserProducts] = Json.format
```

Comment:
ARe you missing the type of the `json.format` here? should it be `json.format[yourType]`?